File: cpp/data_structure/node.test.cpp
#include <cassert>
#include <iostream>

#include "node.hpp"

using namespace std;

int main() {
  Node<int> node(100);
  Node<int> node2(101);

  assert(node.getValue() == 100);
  assert(node.toString() == "Node[100]");

  node.setNext(node2);
  assert(node.getNext()->getValue() == 101);

  cout << "Test passed" << endl;

  return 0;
}

File: java/lib/src/test/java/algorithm/data_structure/NodeTest.java
package algorithm.data_structure;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import org.junit.jupiter.api.Test;

public class NodeTest {
  @Test
  void test_node() {
    Node<String> node = new Node<String>("test");
    Node<String> nextNode = new Node<String>("next");

    assertEquals("test", node.getValue());
    assertEquals("next", nextNode.getValue());
    assertNull(node.getNext());

    node.setNext(nextNode);
    assertEquals(nextNode, node.getNext());

    assertEquals("Node[test]", node.toString());
  }
}
